packages/flutter/lib/src/material/data_table.dart

```diff
@@ -830,10 +837,12 @@ class DataTable extends StatelessWidget {
     required bool ascending,
     required MaterialStateProperty<Color?>? overlayColor,
     required MouseCursor? mouseCursor,
+    required MainAxisAlignment? mainAxisAlignment,
   }) {
     final ThemeData themeData = Theme.of(context);
     final DataTableThemeData dataTableTheme = DataTableTheme.of(context);
     label = Row(
+      mainAxisAlignment: mainAxisAlignment ?? MainAxisAlignment.start,
```

Comment:
Good point. Your suggestion made me question if we need `mainAxisAlignment` at all since non-MainAxisAlignment.center are ineffective. 
We could still use `Expanded` and `Flexible` widgets as the document suggest to get the non-MainAxisAlignment.center behaviors. 

What should do you think If i replace `mainAxisAlignment` with `bool` centerTitle value. Something `AppBar` has And I can add the conditions to strictly center the title as you suggested above.

Reply:
I'll update this with `centerTitle` property.